src/__integrations__/pm.ts

```diff
@@ -157,61 +349,46 @@ walletRouter.post('/pp-restapi/v4/wallet', function (req, res) {
           } else if (!Object.prototype.hasOwnProperty.call(decodedReq.data, 'idPagamentoFromEC')) {
             return res.sendStatus(500);
           } else {
-            const obscuredPan = fromNullable(decodedReq.data?.creditCard?.pan as string)
-              .map(myPan => '*'.repeat(myPan?.length - 4) + myPan?.substr(myPan.length - 4))
-              .getOrElse('************4444');
-            return res.json({
-              data: {
-                idWallet: goodIdWallet,
-                type: 'CREDIT_CARD',
-                favourite: false,
-                creditCard: {
-                  id: 48,
-                  holder: decodedReq.data?.creditCard?.holder,
-                  pan: obscuredPan,
-                  expireMonth: decodedReq.data?.creditCard?.expireMonth,
-                  expireYear: decodedReq.data?.creditCard?.expireYear,
-                  brandLogo: 'http://localhost:8080/wallet/assets/img/creditcard/generic.png',
-                  flag3dsVerified: false,
-                  brand: 'OTHER',
-                  hpan: myFake.random.alphaNumeric(64),
-                  onUs: false,
-                },
-                psp: {
-                  id: 8,
-                  idPsp: 'POSTE1',
-                  businessName: 'Poste Italiane',
-                  paymentType: 'CP',
-                  idIntermediary: 'BANCOPOSTA',
-                  idChannel: 'POSTE1',
-                  logoPSP: 'http://pagopa-dev:8080/pp-restapi/v4/resources/psp/8',
-                  serviceLogo: 'http://pagopa-dev:8080/pp-restapi/v4/resources/service/8',
-                  serviceName: 'nomeServizio 02 poste (MOD0)',
-                  fixedCost: [Object],
-                  appChannel: false,
-                  serviceAvailability: 'disponibilitaServizio FRANCESE',
-                  urlInfoChannel: 'http://www.test.sia.eu',
-                  paymentModel: 0,
-                  idCard: 11008,
-                  lingua: 'IT',
-                  codiceAbi: '06220',
-                  isPspOnus: false,
-                  solvedByPan: false,
-                },
-                idPsp: 8,
-                pspEditable: false,
-                onboardingChannel: 'WISP',
-                services: ['FA', 'pagoPA', 'BPD'],
-                isPspToIgnore: false,
-                saved: false,
-                registeredNexi: false,
-              },
-            });
+            return res.json(fakeGoodWallet(decodedReq));
           }
         },
       ),
   );
 });
+
+walletRouter.put('/pp-restapi/v4/wallet/:id', function (req, res) {
+  fromPredicate(
+    (myReq: typeof req) => /Bearer [\d\w]{128}/.test(fromNullable(myReq.headers.authorization).getOrElse('')),
+    identity,
+  )(req).fold(
+    () => res.sendStatus(401),
+    myReq =>
+      WalletRequest.decode(myReq.body).fold(
+        () => res.sendStatus(500),
+
+        decodedReq => {
+          if (
+            // request id {}
```

Comment:
@pasqualespica I can't verify this case, since I have patched the spec to have `data` field required, so if I put an empty payload I get a type error from the compiler